File: Day 1/BFS.java

class Solution
{
    public ArrayList<Integer> bfsOfGraph(int V, ArrayList<ArrayList<Integer>> adj)
    {
        ArrayList<Integer> ans= new ArrayList<>();
        LinkedList<Integer> queue=new LinkedList<>();
        boolean visited[]=new boolean[V];
        queue.add(0);
        ans.add(0);
        visited[0]=true;
        while(!(queue.size()==0)){
            
            int size=queue.size();
            for(int i=0;i<size;i++){
                int cur=queue.removeFirst();
                for(Integer neighbour:adj.get(cur)){
                    if(!visited[neighbour]){
                        ans.add(neighbour);
                        queue.add(neighbour);
                        visited[neighbour]=true;
                    }
                }
            }
        }
        return ans;
    }
}

File: Day 1/BFS.py

class Solution:
    def bfsOfGraph(self, V, adj):
        ans=list()
        visited=[False]*V
        queue=list()
        queue.append(0)
        ans.append(0)
        while(len(queue)!=0):
            size=len(queue)
            
            for i in range(size):
                cur=queue.pop(0)
                for neighbour in adj[cur]:
                    if(visited[neighbour]==False):
                        visited[neighbour]=True
                        ans.append(neighbour)
                        queue.append(neighbour)
                
       
        return ans
